R/summarise_country.R

```diff
@@ -0,0 +1,163 @@
+summarise_sheet_indicator_list <- function(billion) {
+  billion <- rlang::arg_match(billion)
+  indicator_order_b <- indicator_order %>%
+    dplyr::filter(ind %in% billion_ind_codes(billion)) %>%
+    dplyr::arrange(order) %>%
+    dplyr::select(sdg, short_name, medium_name, unit_raw, )
+
+  return(indicator_order_b)
+}
+
+#' Summarise Excel sheet 'data' for HPOP country summary
+#'
+#' `summarise_HPOP_cntry_data` summarises the 'data' sgeet for the HPOP country summary.
+#'
+#' @param iso3 ISO3 codes of country to export.
+#' @param ... Additional arguments passed to `calculate_hpop_contributions` and
+#' `calculate_hpop_billion`. Use if you need to provide additional parameters to
+#' those functions.
+#' @inherit calculate_hpop_contributions params
+#'
+#' @return list of data frames to be used in the 'data' sheet of HPOP country summary
+#' export
+#'
+#' @export
+summarise_HPOP_cntry_data <-
+  function(df,
+           iso3,
+           start_year = 2018,
+           end_year = 2019:2023,
+           ...) {
+    assert_mart_columns(df)
```

Comment:
Could use `assert_mart_columns()` here, but are all of those columns required? You could just use the `assert_columns()` function if it's a different subset for billions or not too many necessary.

Reply:
My thinking was to control that the input were in an appropriate format, so from `load_billion_data()`. Checking only column name doesn't really achieve either but it was a short cut.
Indeed, not sure anymore if it's necessary to do that as there is actually a narrower set of columns that are needed. I will do that then (also gives the ability to pass function variables rather that the standard xMart columns.